matchms/filtering/remove_peaks_around_precursor_mz.py

```diff
@@ -0,0 +1,34 @@
+import numpy
+from ..Spikes import Spikes
+from ..typing import SpectrumType
+
+
+def remove_peaks_around_precursor_mz(spectrum: SpectrumType, mz_tolerance: float = 17) -> SpectrumType:
+
+    """Remove peaks that are within mz_tolerance (in Da) of
+       the precursor mz, exlcuding the precursor peak
+
+    Args:
+    -----
+    spectrum:
+        Input spectrum.
+    mz_tolerance:
+        Tolerance of mz values that are not allowed to lie
+        within the precursor mz. Default is 17 Da.
+    """
+
+    assert mz_tolerance >= 0, "mz_tolerance must be a positive floating point."
+    precursor_mz = spectrum.get("precursor_mz")
+    mzs, intensities = spectrum.peaks
+    new_mzs, new_intensities = mzs, intensities
+    if precursor_mz:
+        for i, mz in enumerate(mzs):
+
+            if abs(precursor_mz-mz) <= mz_tolerance and mz != precursor_mz:
+                new_mzs[i], new_intensities[i] = numpy.nan, numpy.nan
+
+        nans = numpy.isnan(new_mzs)
+        new_mzs, new_intensities = new_mzs[~nans], new_intensities[~nans]
```

Comment:
```suggestion
    if precursor_mz:
        peaks_to_remove = ((numpy.abs(precursor_mz-mzs) <= mz_tolerance) & (mzs != precursor_mz))
        new_mzs, new_intensities = mzs[~peaks_to_remove], intensities[~peaks_to_remove]
```

Reply:
I guess that a numpy way of implementing this might be faster (and makes it a bit simpler code-wise). Should be doing exactly the same thing (I hope).

Reply:
Yeah works just the same. Thanks.